lib/iron_mq/message.rb

```diff
@@ -0,0 +1,92 @@
+require 'iron_mq/entity'
+
+module IronMQ
+  class Message
+    include Entity
+
+    define_fields :id, :body, :reservation_id, :reserved_count,
+                  :subscribers, :headers, :push_statuses
+
+    def initialize(client, queue_name, msg_fields)
+      @client = client
+      if queue_name.nil? || queue_name.empty?
+        fail ArgumentError, 'queue_name must be non-empty String'
+      end
+      @queue_name = queue_name
+      self.entity = msg_fields
+
+      fail ArgumentError, 'msg_fields must contain message ID' unless id
+    end
+
+    def get!
```

Comment:
There are a lot of methods in ruby that mutate without bangs  e.g. Array `pop` or `push`. If we have a "dangerous" method then it would be reasonable to have a “non-dangerous” equivalent method.
Something like:
```ruby
def get
  @client.call_api(:get, path)
end

def get!
  response = @client.call_api(:get, path)
  self.entity = response['message']
  response
end
```
But I think we are overusing !-methods anyway